### PR TITLE
Fix: duplicate task ids

### LINKED ISSUE #838: Fix duplicate task IDs
## Description
Fix: we often create new tasks in a batch. This is currently a problem because as the backend receives all these requests, it assigns the tasks IDs simultaneously, which causes multiples tasks to have the same ID (where every ID should be unique, and should, when adding a new task, be the last task ID + 1).

## PROPOSED TODO

The issue is located here: `app\admin\quests\dashboard\[questId]\page.tsx`

- [ ] Rather than using `addedTasks.map` (which doesn't wait for the current request to get answered before doing the next one), we should use a for loop. That way, each request will be processed separately, preventing simultaneous ID assignment.

```ts
const handleAddTasks = useCallback(async (addedTasks: StepMap[]) => {
    const taskPromises = addedTasks.map(async (step) => {
      if (step.type === "Quiz") {
        ...
      
      if (step.type === "TwitterFw") {
        ...
      } else if (step.type === "TwitterRw") {
      ...
```

### PR DESCRIPTION
**Pull request by deeseeker**
# Pull Request type

Please add the labels corresponding to the type of changes your PR introduces:

- Bugfix

Resolves: #838

## Summary by CodeRabbit

- **New Features**
	- Enhanced task management with improved error handling for adding tasks.

- **Style**
	- Standardized quotation marks from double to single across various components.
	- Adjusted formatting for improved code readability without altering functionality.

- **Chores**
	- Updated `eslint` dependency version to improve code quality tools.

### CODE REVIEW

**Review by Marchand-Nicolas — CHANGES_REQUESTED** [review]
Hello,

Could you disable your linter for this PR please ? It isn't configured like ours, and it creates a loooot of changes (`;` are removed, `"` are replaced by `'`, etc, but we want to keep it as it was).
image

**Comment by deeseeker** [comment]
haha sorry, I forgot to run the lint. I will make the necessary changes now.

File: app/admin/quests/dashboard/[questId]/page.tsx [hunk 1]
```diff
@@ -439,176 +439,184 @@ export default function Page({ params }: QuestIdProps) {
...
+    for (const step of removedTasks) {
+      try {
+        await AdminService.deleteTask({
+          id: step.data.id,
+        });
+      } catch (error) {
+        console.error("Error deleting task:", error);
+      }
+    }
   }, []);
```
Comment (Marchand-Nicolas) [inline app/admin/quests/dashboard/[questId]/page.tsx]: This part is not needed, only the handleAddTasks needs the for loop

File: app/admin/quests/dashboard/[questId]/page.tsx [hunk 2]
```diff
@@ -439,176 +439,184 @@ export default function Page({ params }: QuestIdProps) {
...
 
   const handleUpdateTasks = useCallback(async (updatedSteps: StepMap[]) => {
-    const taskPromises = updatedSteps.map(async (step) => {
+    for (const step of updatedSteps) {
       if (step.type === "Quiz") {
-        await AdminService.updateQuiz({
-          id: step.data.id,
-          name: step.data.quiz_name,
-          desc: step.data.quiz_desc,
```
Comment (Marchand-Nicolas) [inline app/admin/quests/dashboard/[questId]/page.tsx]: Same, we don't need to use the for loop here, you can keep the old way

**Review by Marchand-Nicolas — CHANGES_REQUESTED** [review]
Please remove some changes, we only want a for loop when adding tasks. After that we can merge!

File: app/admin/quests/dashboard/[questId]/page.tsx (hunk 1 above)
Comment (deeseeker) [reply app/admin/quests/dashboard/[questId]/page.tsx]: Okay

**Review by Marchand-Nicolas — APPROVED** [review]
Lgtm!

**Pull request merged**